### PR TITLE
ceph: Upgrade guide from v1.3 to v1.4

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
The upgrade guide is now updated for upgrades from v1.3 to v1.4. The v1.2 upgrade yamls are removed and the docs are cleaned up from 1.3 upgrade steps that are no longer necessary.

For several releases, Rook has also had unnecessary aggregated clusterroles. Now we remove them and stick with the normal clusterroles. The aggregate clusterrole were designed for the scenario where the rules are not completely owned by one component. Since Rook owns all of the RBAC for its components, the aggregate rules can be removed and simplify certain issues around upgrades.

Resolves: #5965 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

// known CI issues
[skip ci]

### CODE REVIEW

File: tests/framework/installer/ceph_manifests.go [hunk 1]
```diff
@@ -816,20 +816,6 @@ metadata:
   labels:
     operator: rook
     storage-backend: ceph
-aggregationRule:
```
Comment (BlaineEXE) [inline tests/framework/installer/ceph_manifests.go:819]: It seems like the `ceph_manifests.go` are changing less than the manifests themselves. Should there be these changes applied for the upgrade test also?

File: Documentation/ceph-upgrade.md [hunk 2]
```diff
@@ -324,109 +340,101 @@ time without compatibility support and without prior notice.
...
 
 > Automatically updated if you are upgrading via the helm chart
 
-The largest portion of the upgrade is triggered when the operator's image is updated to `v1.3.x`.
+The largest portion of the upgrade is triggered when the operator's image is updated to `v1.4.x`.
 When the operator is updated, it will proceed to update all of the Ceph daemons.
 
 ```sh
-kubectl -n $ROOK_SYSTEM_NAMESPACE set image deploy/rook-ceph-operator rook-ceph-operator=rook/ceph:v1.3.0
+kubectl -n $ROOK_SYSTEM_NAMESPACE set image deploy/rook-ceph-operator rook-ceph-operator=rook/ceph:v1.4.0
```
Comment (leseb) [inline Documentation/ceph-upgrade.md:377]: Should we go with  `ROOK_NAMESPACE` instead? Or is it for legacy support?

File: tests/framework/installer/ceph_manifests.go (hunk 1 above)
Comment (travisn) [reply tests/framework/installer/ceph_manifests.go:819]: Yes, I do need to update the upgrade test as well

File: Documentation/ceph-upgrade.md (hunk 2 above)
Comment (travisn) [reply Documentation/ceph-upgrade.md:377]: Yes that's for legacy support. This var would be the same as ROOK_NAMESPACE if it's a more recent cluster.

File: tests/framework/installer/ceph_manifests.go (hunk 1 above)
Comment (travisn) [reply tests/framework/installer/ceph_manifests.go:819]: @BlaineEXE The upgrade test is updated now as well

File: Documentation/ceph-upgrade.md [hunk 3]
```diff
@@ -98,46 +100,57 @@ volumesnapshotclass.snapshot.storage.k8s.io "csi-rbdplugin-snapclass" deleted
...
+the Alpha CRD
 
-The procedure to migrate the OSDs requires raw devices or partitions be available as mentioned in the [Ceph Prerequisites](ceph-prerequisites.md).
-After you have available storage devices available in your cluster, do the following:
-1. Configure OSDs on the new devices using the [Storage selection settings](https://rook.github.io/docs/rook/v1.3/ceph-cluster-crd.html#storage-selection-settings).
-   If you have specified `useAllDevices: true`, you may need to restart the operator in order to trigger creation of new OSDs on new devices.
-2. Follow the [OSD Removal Guide](https://rook.github.io/docs/rook/v1.3/ceph-osd-mgmt.html#remove-an-osd) to remove the legacy OSDs.
-   - Before removing each OSD, ensure the PGs are all `active+clean` before continuing with the next OSD to ensure your data is safe.
+```bash
+$ kubectl delete crd volumesnapshotclasses.snapshot.storage.k8s.io volumesnapshotcontents.snapshot.storage.k8s.io volumesnapshots.snapshot.storage.k8s.io
```
Comment (leseb) [inline Documentation/ceph-upgrade.md:126]:
```suggestion
kubectl delete crd volumesnapshotclasses.snapshot.storage.k8s.io volumesnapshotcontents.snapshot.storage.k8s.io volumesnapshots.snapshot.storage.k8s.io
```

It's easier to copy :)

File: Documentation/ceph-upgrade.md [hunk 4]
```diff
@@ -98,46 +100,57 @@ volumesnapshotclass.snapshot.storage.k8s.io "csi-rbdplugin-snapclass" deleted
...
+As we have `v1alpha1` CRD created in our kubernetes cluster, we need to delete
+the Alpha CRD
 
-The procedure to migrate the OSDs requires raw devices or partitions be available as mentioned in the [Ceph Prerequisites](ceph-prerequisites.md).
-After you have available storage devices available in your cluster, do the following:
-1. Configure OSDs on the new devices using the [Storage selection settings](https://rook.github.io/docs/rook/v1.3/ceph-cluster-crd.html#storage-selection-settings).
-   If you have specified `useAllDevices: true`, you may need to restart the operator in order to trigger creation of new OSDs on new devices.
-2. Follow the [OSD Removal Guide](https://rook.github.io/docs/rook/v1.3/ceph-osd-mgmt.html#remove-an-osd) to remove the legacy OSDs.
-   - Before removing each OSD, ensure the PGs are all `active+clean` before continuing with the next OSD to ensure your data is safe.
+```bash
```
Comment (leseb) [inline Documentation/ceph-upgrade.md:125]:
```suggestion
```console
```

File: Documentation/ceph-upgrade.md [hunk 5]
```diff
@@ -98,46 +100,57 @@ volumesnapshotclass.snapshot.storage.k8s.io "csi-rbdplugin-snapclass" deleted
...
-If you have legacy OSDs as described in the previous section, it is recommended to migrate the OSDs before the upgrade,
-although the migration is not strictly required before upgrade. If you proceed with the upgrade to v1.3 the OSDs will
-continue running. However, they will no longer be managed by Rook. The OSDs will never again be updated when you update
-Ceph or any other settings in your CephCluster CR.
+As we are updating the snapshot resources from `Alpha` to `Beta` we need to
+delete the old `alphav1` snapshot CRD created by external-snapshotter sidecar container
+
+Check if we have any `v1alpha1` CRD created in our kubernetes cluster
+
+```bash
```
Comment (leseb) [inline Documentation/ceph-upgrade.md:110]:
```suggestion
```console
```

File: Documentation/ceph-upgrade.md [hunk 6]
```diff
@@ -472,119 +475,51 @@ Determining when the Ceph has fully updated is rather simple.
...
-
-```sh
-kubectl apply -f upgrade-from-v1.3-apply.yaml
+```yaml
+  ROOK_CSI_CEPH_IMAGE: "quay.io/cephcsi/cephcsi:v3.0.0"
+  ROOK_CSI_REGISTRAR_IMAGE: "quay.io/k8scsi/csi-node-driver-registrar:v1.2.0"
+  ROOK_CSI_PROVISIONER_IMAGE: "quay.io/k8scsi/csi-provisioner:v1.6.0"
+  ROOK_CSI_SNAPSHOTTER_IMAGE: "quay.io/k8scsi/csi-snapshotter:v2.1.1"
+  ROOK_CSI_ATTACHER_IMAGE: "quay.io/k8scsi/csi-attacher:v2.1.0"
+  ROOK_CSI_RESIZER_IMAGE: "quay.io/k8scsi/csi-resizer:v0.4.0"
```
Comment (leseb) [inline Documentation/ceph-upgrade.md:512]:
```suggestion
ROOK_CSI_CEPH_IMAGE: "quay.io/cephcsi/cephcsi:v3.0.0"
ROOK_CSI_REGISTRAR_IMAGE: "quay.io/k8scsi/csi-node-driver-registrar:v1.2.0"
ROOK_CSI_PROVISIONER_IMAGE: "quay.io/k8scsi/csi-provisioner:v1.6.0"
ROOK_CSI_SNAPSHOTTER_IMAGE: "quay.io/k8scsi/csi-snapshotter:v2.1.1"
ROOK_CSI_ATTACHER_IMAGE: "quay.io/k8scsi/csi-attacher:v2.1.0"
ROOK_CSI_RESIZER_IMAGE: "quay.io/k8scsi/csi-resizer:v0.4.0"
```

**Pull request merged**